src/XProfile/VirtoCommerce.XProfile/Authorization/ProfileAuthorizationHandler.cs

```diff
@@ -127,6 +127,11 @@ protected override async Task HandleRequirementAsync(AuthorizationHandlerContext
             else if (context.Resource is UpdateOrganizationCommand updateOrganizationCommand && currentContact != null)
             {
                 result = currentContact.Organizations.Contains(updateOrganizationCommand.Id);
+            }
+            else if (context.Resource is UpdateMemberDynamicPropertiesCommand)
```

Comment:
since you haven't used any permissions for auth checking, returning true will allows to anyone updating dynamic properties for any contact, this is the major security vulnerability. 

Please restrict this operation with rules:

A user can update dynamic properties for self contact (which his security account is associated with) 

A user can update dynamic properties for the organization only if has the CustomerModule.Core.ModuleConstants.Security.Permissions.Update permission


(we need a call to discuss how to better implement this)

Reply:
done